LeanKit-Labs/node-hilo
# Bumped deps, added types to support typescript and javascript intellisense

### Short description of the work completed

> This updates to node 16, bumps dependencies, and adds a type definition file.

FYI- there are what error messages generated by `processhost` when running tests. Processhost is dev-only and the failures do not prevent the tests from passing, so I think it's safe to ignore them. If you see a way to prevent them altogether, let me know!

In order to provide full intellisense report, the default factory function has been moved to a named function (`redisFactory`). This is a breaking change; consuming services will require a small change when updating.

### Steps to test (if not obvious)

1. npm link in automation-service
2. remove `node-hilo` definition from types folder
3. verify types are loaded and working as expected
4. verify functionality and intellisense in another js service

### For Reviewer Use Only

- [ ] Is the version upgrade path clear for this change? (breaking vs minor vs patch)
- [ ] Verify PR is being merged into an upstream feature branch. (NOT DEVELOP)
